app/controllers/application_controller.rb

```diff
@@ -1,3 +1,20 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  helper_method :logged_in?, :current_user
+
+  def logged_in?
+    !!current_user
+  end
+
+  def current_user
+    current ||= User.find(session[:user_id]) if session[:user_id]
```

Comment:
Assigning user record to a local variable will do nothing meaningful in this case. That is why it must be instance variable `@current_user`, because this method is most likely getting called _multiple_ times per request and instance variable will cache user record between different calls.